packages/config-loader/src/lib/transform/include.ts

```diff
@@ -79,7 +79,16 @@ export function createIncludeTransform(
         }
       case '$env':
         try {
-          return { applied: true, value: await env(includeValue) };
+          const value = await env(includeValue);
+
+          // Handle boolean type coercion consistently.
+          if (value && /^(?:y|yes|true|1|on)$/i.test(value)) {
+            return { applied: true, value: true };
+          } else if (value && /^(?:n|no|false|0|off)$/i.test(value)) {
+            return { applied: true, value: false };
+          }
```

Comment:
Perhaps instead we could use [`safeJsonParse()`](https://github.com/backstage/backstage/blob/ad32e7c9408d7cfd30908192bf7cdc4208f8262f/packages/config-loader/src/lib/env.ts#L97), which is how `APP_CONFIG_*` coercion happens.

...This would result in coercion for other types as well, though.